packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_loan_payback.test.ts

```diff
@@ -0,0 +1,760 @@
+    }, bobColScript)
+
+    // Ensure the created txn is correct
+    const outs = await sendTransaction(bob.container, txn)
+    expect(outs[0].value).toStrictEqual(0)
+    expect(outs[1].value).toBeLessThan(10)
+    expect(outs[1].value).toBeGreaterThan(9.999)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await bProviders.getAddress())
+
+    // Ensure you don't send all your balance away
+    const prevouts = await bProviders.prevout.all()
+    expect(prevouts.length).toStrictEqual(1)
+    expect(prevouts[0].value.toNumber()).toBeLessThan(10)
+    expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
+
+    await bob.generate(1)
+
+    const vaultAfter = await bob.container.call('getvault', [bobVaultId])
+    expect(vaultAfter.loanAmounts).toStrictEqual([])
+    expect(vaultAfter.interestAmounts).toStrictEqual([])
+    expect(vaultAfter.interestValue).toStrictEqual(0)
+    expect(vaultAfter.loanValue).toStrictEqual(0)
+    expect(vaultAfter.collateralRatio).toStrictEqual(-1)
+    expect(vaultAfter.informativeRatio).toStrictEqual(-1)
+
+    const bobColAccAfter = await bob.rpc.account.getAccount(bobColAddr)
+    expect(bobColAccAfter).toStrictEqual(['4.99990868@TSLA']) // 45 - 40.00004566
+  })
+
+  it('should loanPayback partially', async () => {
+    const burnInfoBefore = await bob.container.call('getburninfo')
+    expect(burnInfoBefore.paybackburn).toStrictEqual(undefined)
+
+    const bobColAccBefore = await bob.container.call('getaccount', [bobColAddr])
+    expect(bobColAccBefore).toStrictEqual(['40.00000000@TSLA'])
+
+    const vaultBefore = await bob.container.call('getvault', [bobVaultId])
+    expect(vaultBefore.collateralValue).toStrictEqual(15000) // DFI(10000) + BTC(1 * 10000 * 0.5)
+    expect(vaultBefore.loanAmounts).toStrictEqual(['40.00002283@TSLA']) // 40 + totalInterest
+    expect(vaultBefore.interestAmounts).toStrictEqual(['0.00002283@TSLA'])
+    expect(vaultBefore.loanValue).toStrictEqual(80.00004566) // loanAmount * 2 (::1 TSLA = 2 USD)
+    expect(vaultBefore.interestValue).toStrictEqual(0.00004566)
+    expect(vaultBefore.collateralRatio).toStrictEqual(18750) // 15000 / 80.00004566 * 100
+    expect(vaultBefore.informativeRatio).toStrictEqual(18749.98929844)
+
+    {
+      const interests = await bob.rpc.loan.getInterest('scheme')
+      const height = await bob.container.getBlockCount()
+      const tslaInterestPerBlock = (netInterest * 40) / (365 * blocksPerDay) //  netInterest * loanAmt / 365 * blocksPerDay
+      const tslaInterestTotal = tslaInterestPerBlock * (height - tslaLoanHeight)
+      expect(interests[0].interestPerBlock.toFixed(8)).toStrictEqual(tslaInterestPerBlock.toFixed(8))
+      expect(interests[0].totalInterest.toFixed(8)).toStrictEqual(tslaInterestTotal.toFixed(8))
+    }
+
+    await fundEllipticPair(bob.container, bProviders.ellipticPair, 10)
+    const bobColScript = P2WPKH.fromAddress(RegTest, bobColAddr, P2WPKH).getScript()
```

Comment:
```
await fundEllipticPair(bob.container, bProviders.ellipticPair, 10)
const bobColScript = P2WPKH.fromAddress(RegTest, bobColAddr, P2WPKH).getScript()
```

Is that possible to move this 2 lines to beforeEach? So you do not need to duplicate the code in every test item